### DISCUSSION TITLE
Why do we need to specify both sides of the road for bike lanes and shoulders?

### DISCUSSION THREAD

**Discussion by endolith** [streetcomplete/StreetComplete#4630] [Feedback]
In the bike lane and shoulder quests, it asks for both sides of the road, and doesn't let you submit just one side:

![Screenshot_20221105-120808_StreetComplete](https://user-images.githubusercontent.com/58611/201708910-2f41dcef-d31d-4ade-baf7-13385ee3d6b7.jpg)

But I can't always see the other side of the road, if it's wide and I'm riding in car, etc.  So we should just be able to submit one half and leave the other half as question marks.

**Comment by mnalis** [comment]
Of of the reasons is likely because SC is [geared toward pedestrians](https://github.com/streetcomplete/StreetComplete/pull/4188) (and not when driving in vehicle), see  [quest guidelines](https://github.com/streetcomplete/StreetComplete/blob/master/QUEST_GUIDELINES.md)  _":detective: Publicly accessible on foot"_

Other one is technical. If you answered only one side, **one** of the following must happen:
- **the timestamp of OSM element is taken into account**, so the quest is not asked (of you, or anyone else) ever again (or at least until possible resurvey period kicks in several years later). This is obviously bad result.
- **the timestamp is ignored**, so the quest is always asked _unless_ both sides are answered. That means, as soon as you confirm the quest, it pops back up right away (as only one side is answered). That goes against how UI should work (when quest is answered it is answered for good), i.e. _":thinking: No unanswerable quests"_ , so again bad result.

Thus, you have to answer both sides.

**Reply by matkoniecz** [reply]
Also, adding support for answering one side only is more complicated. See https://github.com/streetcomplete/StreetComplete/issues/4634

**Reply by endolith** [reply]
> Of of the reasons is likely because SC is https://github.com/streetcomplete/StreetComplete/pull/4188 (and not when driving in vehicle)

That would make it even _less_ likely that I could see what's on the other side of the road, though.
